### ISSUE TITLE
complete -e doesn't

### ISSUE THREAD

**Issue by zanchey** [fish-shell/fish-shell#380]
The documentation for complete says:

"When erasing completions, it is possible to either erase all completions for a specific command by specifying complete -e -c COMMAND, or by specifying a specific completion option to delete by specifying either a long, short or old style option."

The former doesn't work, but that latter does.

```
> complete -c foo -l bar
> foo --b<TAB>ar
> complete -c foo -e
> foo --b<TAB>ar
> complete -e -c foo
> foo --b<TAB>ar
> complete -c foo -e -l bar
> foo --b<TAB><BELL>
```

**Comment by scooter-dangle** [comment]
Wouldn't just plain
```
complete --command gco --arguments "(complete --do-complete='git checkout ')" --exclusive --authoritative
```
work? Is that sub-optimal due to having to open a sub-shell whenever the completion is run?

**Comment by scooter-dangle** [comment]
I had earlier installed fish from master to get completion erasure, but I thought it was included in 2.1.2, which I used brew to install when it came out. The fix doesn't seem to be present...any word on which release will include it?

**Comment by zanchey** [comment]
It will not be in 2.1.2 - that was a bugfix on the 2.1 branch. It's included in the upcoming 2.2 release.

**Comment by scooter-dangle** [comment]
gotcha...thanks for the response
